DOR-89 Frotend > Crear Formulario de registro de libro

DodanimJR opened the pull request:
https://isc-ads.atlassian.net/browse/DOR-89

DodanimJR committed:
Line end CRLF to LF correction

DodanimJR committed:
Added Book Service

DodanimJR committed:
Book service Correction

DodanimJR committed:
Page Structure added

DodanimJR committed:
Page css added

DodanimJR committed:
Added Page Styles

DodanimJR committed:
Added DatePicker Component

DodanimJR committed:
Added Required  Dependencies

DodanimJR committed:
Primary Render

DodanimJR committed:
Button Added

DodanimJR committed:
Final page structure 

Select component added, needs to be fixed

DodanimJR committed:
ISBN correction

DodanimJR committed:
Select corrected

DodanimJR committed:
It gets all values!

DodanimJR committed:
Works with happy path!

DodanimJR committed:
DONE - Create book

Create book finished with error handling!

wmartzh commented:
If there an error on data form does't catch it 

![image](https://user-images.githubusercontent.com/26823527/228107817-864fe30c-4b9f-4521-8b4d-20de40992338.png)

wmartzh commented:
Picture should be optional
![image](https://user-images.githubusercontent.com/26823527/228108517-f4eb7ec5-2c8a-4a15-ac5a-34ec41a4d168.png)

wmartzh commented:
Form should reset after success message

![image](https://user-images.githubusercontent.com/26823527/228108645-8e2aacfd-82dc-4ec0-95d1-ec08736ce0bb.png)

wmartinezh commented on src/services/book.js:
```diff
@@ -0,0 +1,51 @@
+
+export async function collectBookData(method) {
+  try {
+    const book = {
+      title: document.getElementById('BookTitle').value,
+      author: document.getElementById('BookAuthor').value,
+      isbn: document.getElementById('BookISBN').value,
+      category: document.getElementById('BookCategory').innerHTML.toString(),
+      year: document.getElementById('BookYear').getAttribute('value'),
+      picture: document.getElementById('BookPicture').value,
+      editorial: document.getElementById('BookEditorial').value,
+      language: document.getElementById('BookLanguage').innerHTML.toString()
+    };
```
Use react state to capture data, all this function should on the form component, only backend request shold be called from outsied the component

wmartinezh commented on src/pages/books/RegisterBook.jsx:
```diff
@@ -1,7 +1,105 @@
+            <Input
+              height="44px"
+              width="100%"
+              colorHover="#394c73"
```
MUI doesn't have `colorHover` check the doc [Inputs](https://mui.com/material-ui/api/input/)

DodanimJR committed:
WIP - PR Corrections

Added Model validation with JOI,
Swapped functionality to form component instead of book service.
Added models folder
Corrected Alert error Handling

DodanimJR committed:
WIP - commit to my desktop haha

DodanimJR committed:
WIP - PR Correcitions

File uploader ready and alerts corrected, Doubts using the states with the other inputs, i don't get why?

wmartzh committed:
Merge branch 'main' into DOR-89-Frotend-Crear-Formulario-de-registro-de-libro
